src/vector_index.c

```diff
@@ -121,16 +121,16 @@ IndexIterator *NewVectorIterator(QueryEvalCtx *q, VectorQuery *vq, IndexIterator
     }
     case VECSIM_QT_RANGE: {
       if ((dim * VecSimType_sizeof(type)) != vq->range.vecLen) {
-        QueryError_SetErrorFmt(q->status, QUERY_EINVAL,
+        QueryError_SetWithoutUserDataFmt(q->status, QUERY_EINVAL,
```

Comment:
Nitpick - should we be consistent with https://github.com/RediSearch/RediSearch/pull/5719#discussion_r1998550183
and consider actual and expected vector sizes as user data?